### PR TITLE
feat(addon): #417 infinite scroll

### PR DESCRIPTION
**Pull request by k88hudson**
This patch adds "infinite scrolling" to the Timeline view:

* `TimelineHistory` and `TimelineBookmarks` are mostly refactored into a unified `TimelineFeed` component, which contains most of the layout and loading behaviour
* A new `skipPreview` property which allows the request to used cached data if it exists, but bypass the async task to request new items from embedly. This is so infinite scrolls won't create too many extra (mostly unecessary) requests.

Infinite scroll is debounced by `100ms`. There's also a bit of a weird hack to restore the scroll position to the top on a page reload – it's commented.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/activity-stream/751)

### CODE REVIEW

**Comment by k88hudson** [comment]
@rlr or @oyiptong R?

File: content-src/components/TimelinePage/TimelineBookmarks.js [hunk 1]
```diff
@@ -1,43 +1,18 @@
...
-        hidden={!props.Bookmarks.canLoadMore || !props.Bookmarks.rows.length}
-        onClick={this.getMore}
-        label="See more activity"/>
-    </div>);
+    return (<TimelineFeed
+      loadMoreAction={RequestMoreBookmarks}
+      dateKey={"bookmarkDateCreated"}
+      pageName={"TIMELINE_BOOKMARKS"}
+      Feed={props.Bookmarks}
+    />);
```
Comment (rlr) [inline content-src/components/TimelinePage/TimelineBookmarks.js]: berry nice refaktor

File: content-test/components/TimelinePage.test.js [hunk 2]
```diff
@@ -45,81 +49,174 @@ describe("Timeline", () => {
...
+        setup({loadMoreAction: () => {
+          throw new Error("Should not call loadMore");
+        }});
+        instance.windowHeight = 200;
+        instance.handleScroll({scrollTop: 0, scrollHeight: 400});
+      });
+      it("should loadMore if the scrollTop is past the threshold", done => {
+        setup({loadMoreAction: () => done()});
+        instance.windowHeight = 200;
+        const scrollTop = 200 + SCROLL_TOP_OFFSET - INFINITE_SCROLL_THRESHOLD + 1;
```
Comment (rlr) [inline content-test/components/TimelinePage.test.js]: so much math for a Friday afternoon 💭

**Comment by rlr** [comment]
played with it a bunch and WFM. pretty speedy at least in my small test profile! 🚀 R+

File: content-src/components/TimelinePage/TimelineBookmarks.js (hunk 1 above)
Comment (k88hudson) [inline content-src/components/TimelinePage/TimelineBookmarks.js]: 🍓

**Pull request merged**